fiberplane-ci/src/commands/versions.rs

```diff
@@ -0,0 +1,341 @@
+use crate::utils::{toml_patcher::TomlPatcher, toml_query::TomlNode};
+use crate::TaskResult;
+use anyhow::Context;
+use clap::Parser;
+use std::{fs, path::Path};
+use taplo::dom::KeyOrIndex;
+use thiserror::Error;
+
+/// Parameters intended to be set by the xtask implementation, rather than the
+/// CLI user.
+pub struct SetVersionParams<P>
+where
+    P: AsRef<Path>,
+{
+    /// Path of the Cargo file in which to update the version numbers.
+    pub cargo_path: P,
+
+    /// Use this flag if you want the workspace version to be patched as well.
+    pub patch_workspace: bool,
+}
+
+#[derive(Parser)]
+pub struct SetVersionArgs {
+    /// Name of the dependency to bump.
+    #[clap(long, short)]
+    pub dependency: String,
+
+    /// The new version of the dependency.
+    #[clap(long, short)]
+    pub version: String,
+}
+
+pub fn set_version<P>(params: &SetVersionParams<P>, args: &SetVersionArgs) -> TaskResult
+where
+    P: AsRef<Path>,
+{
+    let cargo_toml =
+        fs::read_to_string(params.cargo_path.as_ref()).with_context(|| "Cannot read Cargo file")?;
```

Comment:
prefer using the `.context` when its just a string literal so it gets eagerly evaluated instead of lazily

```suggestion
        fs::read_to_string(params.cargo_path.as_ref()).context("Cannot read Cargo file")?;
```